man/org.freedesktop.sysupdate1.xml

```diff
@@ -0,0 +1,442 @@
+      is specified, that is the version that gets installed. Otherwise, the latest version is installed.
+      Unlike all the other methods in this interface, <function>Update()</function> does not wait for its
+      job to complete. Instead, it returns the job's numeric ID and object path as soon as the job begins,
+      so that the caller can listen for progress updates or cancel the operation. This method also returns
+      the version the target will be updated to, for cases where no version was specified by the caller. This
+      method pulls both metadata and payload data from the network. Listen for the Manager's
+      <function>JobRemoved()</function> signal to detect when the job is complete.</para>
+
+      <para><function>Vacuum()</function> deletes old installed versions of this target to free up space.
+      It returns the number of instances that have been deleted.</para>
+
+      <para><function>GetAppStream()</function> returns a list of HTTP/HTTPS URLs to this target's <ulink url="https://wwww.freedesktop.org/software/appstream/docs/chap-CatalogData.html">appstream catalog</ulink>
+      XML files. If this target has no appstream catalogs, the method will return an empty list. These
+      catalog files can be used by software centers (such as GNOME Software or KDE Discover) to present rich
+      metadata about the target, including a display name, changelog, icon, and more. The returned catalogs
+      will include <ulink url="https://systemd.io/APPSTREAM_BUNDLE">special metadata</ulink> to allow the
+      software center to correctly associate the catalogs with this target.</para>
+
+      <para><function>GetVersion()</function> returns the current version of this target, if any. The current
+      version is the newest version that is installed. Note that this isn't necessarily the same thing as the
+      booted or currently-in-use version of the target.  For example, on the host system the booted version
+      is the current version most of the time, but if an update is installed and pending a reboot it will
+      become the current version instead.  You can query the booted version of the host system via
+      <varname>IMAGE_VERSION</varname> in <filename>/etc/os-release</filename>. If the target has no current
+      version, the function will return an empty string.</para>
+
+    </refsect2>
+
+    <refsect2>
+      <title>Properties</title>
+
+      <para>The <varname>Class</varname> property exposes the class of this target, which describes
+      where it was enumerated. Possible values include: <literal>machine</literal> for containers and
+      virtual machines managed by
+      <citerefentry><refentrytitle>systemd-machined.service</refentrytitle><manvolnum>8</manvolnum></citerefentry>,
+      <literal>portable</literal> for <ulink url="https://systemd.io/PORTABLE_SERVICES">portable services</ulink>,
+      <literal>extension</literal> for system extensions managed by
```

Comment:
Either confext+systext should be managed as one, and then called 'extesion', or separately, and then call this 'sysext'

Reply:
looks like in the code this is "sysext", so I guess the doc should be updated to reflect that

Reply:
Hmm it's called sysext in the code because Lennart asked us to rename it. However, confexts could probably work too?

If it's possible to separate the two at the image discovery level, I would.

Though with the optional features work, it's unclear that it should be possible to update sysexts this way anymore. Maybe the fix is to just not expose extensions through the service right now? Or if we do expose them, only expose confexts. Or should those be components? Idk - something we need to think about